src/cdomains/valueDomain.ml

```diff
@@ -288,7 +288,7 @@ struct
         | TInt (ik,_) ->
           `Int (ID.cast_to ?torg ik (match v with
               | `Int x -> x
-              | `Address x when AD.equal x AD.null_ptr -> ID.cast_to ?torg (ptr_ikind ()) @@ ID.of_int Int64.zero
+              | `Address x when AD.equal x AD.null_ptr -> ID.cast_to ?torg (ptr_ikind ()) @@ ID.of_int (Cilfacade.ptrdiff_ikind ()) Int64.zero
```

Comment:
What is the reason behind making a `0` of `Cilfacade.ptrdiff_ikind ()` and then casting it to `ptr_ikind ()`?